netmiko/adtran/adtran.py

```diff
@@ -0,0 +1,40 @@
+import time
+import re
+from netmiko.cisco_base_connection import CiscoBaseConnection
+from netmiko import log
+
+
+class AdtranOSBase(CiscoBaseConnection):
+    def session_preparation(self):
+        """Prepare the session after the connection has been established."""
+        self.ansi_escape_codes = True
+        self._test_channel_read()
+        self.set_base_prompt()
+        self.disable_paging(command="terminal length 0")
+        # Clear the read buffer
+        time.sleep(0.3 * self.global_delay_factor)
+        self.clear_buffer()
+
+    def strip_ansi_escape_codes(self, string_buffer):
+        """
+        Huawei does a strange thing where they add a space and then add ESC[1D
+        to move the cursor to the left one.
+
+                Adtran also encounters this same issue like Huawei when executing long commands...
+                This function works as intended with Adtran also (Taken from Huawei)
+
+        The extra space is problematic.
+        """
+        code_cursor_left = chr(27) + r"\[\d+D"
+        output = string_buffer
+        pattern = rf" {code_cursor_left}"
+        output = re.sub(pattern, "", output)
+
+        log.debug("Stripping ANSI escape codes")
+        log.debug(f"new_output = {output}")
+        log.debug(f"repr = {repr(output)}")
+        return super().strip_ansi_escape_codes(output)
```

Comment:
Does Adtran have this behavior also. This was a Huawei oddity we observed recently.

Reply:
My apologies, I thought it was the exact same issue but it's not exactly the same.

When you write long commands, the prompt will backspace so this:
```
ol03.pag-lab.lon(config-shaper my-test-shaper 1/1)#per interface muxponder-highspeed 1/1/1
```
becomes like this
```
ol03.pag-lab.lon(config-shaper my-test-shaper 1/1)#$face muxponder-highspeed 1/1/1
```

Debug log:
```
DEBUG:netmiko:In strip_ansi_escape_codes
DEBUG:netmiko:repr = 'per interface muxponder-\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08$er interface muxponder-h \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08$r interface muxponder-hi  \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08$ inte'
DEBUG:netmiko:Stripping ANSI escape codes
```

Reply:
@BSpendlove Hmmm...that really sounds like an issue/behavior that setting the terminal width should fix. How did you set and test the terminal width?

There is this method you can call in Netmiko

```
strip_backspaces()
```

Here is an example of calling it:

`output = self.strip_backspaces(output)`

Reply:
Yeah I would of hoped setting the terminal width would fix it, I set it to the max and simply tried sending creating a shaper. I've got a bit further, the commands are actually sent now but the returns output is still a bit shady...

```
def _read_channel_expect(self, *args, **kwargs):
        output = super()._read_channel_expect()

        #Strip backspaces for long commands
        output = self.strip_backspaces(output)
        return output
```

```
DEBUG:netmiko:shaper my-test-shaper 1/1
adtran(config-shaper my-test-shaper 1/1)#per interface muxponder-$er interface muxponder-h $r interface muxponder-hi  $ interface muxponder-hig  $interface muxponder-high  $nterface muxponder-highs  $terface muxponder-highsp  $erface muxponder-highspe  $rface muxponder-highspee  $face muxponder-highspeed  $ace muxponder-highspeed   $ce muxponder-highspeed 1  $e muxponder-highspeed 1/  $ muxponder-highspeed 1/1  $muxponder-highspeed 1/1/  $uxponder-highspeed 1/1/1  
adtran(config-shaper my-test-shaper 1/1)#end
```

I'll get their eventually 😸

Reply:
@ktbyers I had set the terminal width to the max (132) and created an EVC map on the adtran device, however I found a command that hits the max length and runs into the issue that I found. I've been able to use the `strip_backspaces()` and get rid of the `\x08` but now just figuring out why I'm hitting another issue 👍